### PR TITLE
ABCD-19 Created DecedentController

### PR DESCRIPTION
**Pull request by brzezin0**

### CODE REVIEW

File: src/main/java/com/app/backend/controller/DecedentController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+
+@RestController
+@RequestMapping("/api/decedent")
+@CrossOrigin("*")
+public class DecedentController {
+
+    @Autowired
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/controller/DecedentController.java:17]: Autowired is unneccesery here. DataSource for example is better. Autowired can be pretty hectic and hard to control

File: src/main/java/com/app/backend/controller/DecedentController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+
+@RestController
+@RequestMapping("/api/decedent")
+@CrossOrigin("*")
+public class DecedentController {
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/controller/DecedentController.java:15]: While writing microservices it is a good practice to not have implementation in the same file. This way the URL is more visible and more code-reading friendly

File: src/main/java/com/app/backend/controller/DecedentController.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Autowired
+    private DecedentRepository decedentRepository;
+
+    @PostMapping("/add")
+    public ResponseEntity<String> addDecedent(@RequestBody Decedent decedent) {
+        ResponseEntity<String> validationResponse = validateDecedent(decedent);
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/controller/DecedentController.java:22]: We should check if we at least started the endpoint. You can simply add System.out.println "Adding started for: " ...)

File: src/main/java/com/app/backend/controller/DecedentController.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            decedentRepository.save(decedent);
+            return new ResponseEntity<>("Decedent added successfully", HttpStatus.CREATED);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Error saving decedent: " + e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    private ResponseEntity<String> validateDecedent(Decedent decedent){
+        if (decedent == null) {
+            return ResponseEntity.status(HttpStatus.NOT_ACCEPTABLE).body("Decedent data is null");
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/controller/DecedentController.java:37]: I'd go with 409 or 422

**Review by Ikar17 — APPROVED** [review]

**Pull request merged**